bot/handlers/governance_handlers.py

```diff
@@ -0,0 +1,210 @@
+
+    try:
+        proposal = get_proposal(int(proposal_id))
+        context.user_data.setdefault('proposals_cache', {})[proposal_id] = {
+            'title': proposal['content']['value']['title']}
+    except Exception as e:
+        logger.error(e)
+        query.edit_message_text(NETWORK_ERROR_MSG)
+        return
+
+    keyboard = [[InlineKeyboardButton(BACK_BUTTON_MSG, callback_data=previous_view)]]
+    my_wallet = context.user_data['proposals_cache'].get('wallet', None)
+    my_vote = None
+
+    if my_wallet:
+        my_vote = get_vote(wallet_addr=my_wallet, proposal_id=proposal_id)
+
+    message = ''
+    if my_wallet and my_vote:
+        message += f'🎉 You voted *{my_vote}* 🎉'
+    else:
+        if votable:
+            keyboard = [
+                [
+                    InlineKeyboardButton('✅ Yes', callback_data=f'vote-{proposal_id}-{MsgVote.YES}'),
+                    InlineKeyboardButton('❌ No', callback_data=f'vote-{proposal_id}-{MsgVote.NO}'),
+                ],
+                [
+                    InlineKeyboardButton('❌❌ No with veto',
+                                         callback_data=f'vote-{proposal_id}-{MsgVote.NO_WITH_VETO}'),
+                    InlineKeyboardButton('🤷 Abstain', callback_data=f'vote-{proposal_id}-{MsgVote.ABSTAIN}'),
+                ],
+                [InlineKeyboardButton(BACK_BUTTON_MSG, callback_data='proposals_show_active')]
+            ]
+
+    message += f"\n\n\n{proposal_to_text(proposal)}"
+
+    query.edit_message_text(message, parse_mode='markdown', reply_markup=InlineKeyboardMarkup(keyboard))
+
+
+def on_vote_option_clicked(update, context):
+    query = update.callback_query
+    _, proposal_id, vote = query.data.split("-")
+    proposal_title = context.user_data['proposals_cache'][proposal_id]['title']
+    wallet = context.user_data['proposals_cache']['wallet']
+    text = f'⚠️ Do you really want to vote *{vote}* on proposal *{proposal_title}*? ⚠️\n'
+
+    if wallet:
+        text += f'Your voting address is *{wallet}*'
+        yes_button = InlineKeyboardButton('YES ✅', callback_data=f'votesend-{proposal_id}-{vote}')
+    else:
+        text += YOU_WILL_BE_REDIRECTED_MSG
+        yes_button = InlineKeyboardButton('YES ✅ (open website)', url=f'{WEBSITE_URL}vote?id={proposal_id}&vote={vote}')
+
+    keyboard = [
+        [yes_button,
+         InlineKeyboardButton(BACK_BUTTON_MSG, callback_data=f'proposal-{proposal_id}-1')]]
+    query.edit_message_text(text, parse_mode='markdown', reply_markup=InlineKeyboardMarkup(keyboard))
+
+
+def on_authorize_voting_clicked(update, _):
+    text = "You are going to grant me the authorization for voting.\n" \
+           "I will *never* vote on your behalf except when you order me to do. 👮\n" \
+           "If you confirm the authorization, you will be able to vote directly from any Telegram client" \
+           " without accessing your wallet at all.\n\n"
```

Comment:
Here users need to fully trust us. Therefore it might make sense to include a link to msgAuthorization function of the docs